### PR TITLE
LOK-2091: Top Nodes & Reachability 

### PR DESCRIPTION
**Pull request by mikewrosey**
## Description
Adds initial top nodes table and reachability doughnut chart.
Adds new alert design

pagination / cleared & total counts / file downloads to come later

screenshot-localhost_3009-2023 11 14-14_47_09

screenshot-localhost_3009-2023 11 14-14_47_23

screenshot-localhost_3009-2023 11 14-14_52_16

screenshot-localhost_3009-2023 11 14-14_52_05

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2091

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/components/Dashboard/DashboardNodeReachability.vue [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+const doughnutCentreText = {
+  id: 'doughnutCentreText',
+  beforeDatasetsDraw(chart: Chart) {
+    const { ctx, data } = chart
+
+    ctx.save()
+
+    const reachable = data.datasets[0].data[0] as number
+    const unreachable = data.datasets[0].data[1] as number
+    const availability = Math.round((reachable / (reachable + unreachable)) * 100)
```
Comment (synqotik) [inline ui/src/components/Dashboard/DashboardNodeReachability.vue:34]: May want to check for divide-by-zero here? Or does the entire expression result in `NaN` in that case?

File: ui/src/store/Views/dashboardStore.ts [hunk 2]
```diff
@@ -22,6 +29,14 @@ export const useDashboardStore = defineStore('dashboardStore', {
...
       await queries.getNetworkTrafficOutMetrics()
       this.totalNetworkTrafficOut = (queries.networkTrafficOut as TsResult).metric?.data?.result[0]?.values || []
+    },
+    async getTopNNodes() {
+      const queries = useDashboardQueries()
+      await queries.getTopNodes()
+      this.topNodes = queries.topNodes
+
+      this.reachability.responding = queries.topNodes.filter((n) => n.avgResponseTime > 0).length
+      this.reachability.unchrachable = queries.topNodes.length - this.reachability.responding
```
Comment (synqotik) [inline ui/src/store/Views/dashboardStore.ts:39]: Should be `this.reachability.unreachable` ?

**Review by synqotik — APPROVED** [review]
Looks good. Two small comments.

File: ui/src/store/Views/dashboardStore.ts (hunk 2 above)
Comment (mikewrosey) [reply ui/src/store/Views/dashboardStore.ts:39]: Yes, will fix that

File: ui/src/components/Dashboard/DashboardNodeReachability.vue (hunk 1 above)
Comment (mikewrosey) [reply ui/src/components/Dashboard/DashboardNodeReachability.vue:34]: Yup, it's NaN in that case

**Pull request merged**